packages/medusa/src/services/inventory.js

```diff
@@ -0,0 +1,73 @@
+import { BaseService } from "medusa-interfaces"
+import { MedusaError } from "medusa-core-utils"
+const fs = require("fs")
+
+class InventoryService extends BaseService {
+  constructor({ manager, productVariantService }) {
+    super()
+
+    /** @private @const {EntityManager} */
+    this.manager_ = manager
+
+    /** @private @const {ProductVariantRepository_} */
+    this.productVariantService_ = productVariantService
+  }
+
+  withTransaction(transactionManager) {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new InventoryService({
+      manager: transactionManager,
+      productVariantService: this.productVariantService_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  /**
+   * Updates the inventory of a variant based on a given adjustment.
+   * @params {string} variantId - the id of the variant to update
+   * @params {number} adjustment - the number to adjust the inventory quantity by
+   * @return {Promise} resolves to the update result.
+   */
+  async adjustInventory(variantId, adjustment) {
+    return this.atomicPhase_(async (manager) => {
+      const variant = await this.productVariantService_.retrieve(variantId)
```

Comment:
I think we need to add a check, that ensures that the variant should actually be updated. If `manage_inventory` is set to false, we do not want to update it right?